torch/export/dynamic_shapes.py

```diff
@@ -698,9 +698,16 @@ def check_same_bounds(dim):
             bounds[dim.__name__] = (dim.min, dim.max)
 
     def update_symbols(tensor, shape):
+        def _create_static_dim(tensor, i, value):
+            return _Dim(
+                f"static_dim_{id(tensor)}_{i}", (int,), {"min": value, "max": value}
```

Comment:
Hmm so this is not what I had in mind. I think it would be better to have a subclass of `_Dim` to represent this case. You can probably define min and max as properties that give you the same constant. Or you could explicitly handle this subclass wherever it appears (e.g., look for `_DerivedDim`). 

In general I don't want to violate the assumption that `min < max`, I know this is an easy way to implement this but it might cause problems with `ValueRanges` assumptions elsewhere in `symbolic_shapes`. Open to be convinced otherwise.